javaslang/src/main/java/javaslang/collection/Queue.java

```diff
@@ -1135,11 +1135,17 @@ public boolean startsWith(Iterable<? extends T> that, int offset) {
         return ofAll(toList().update(index, element));
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     public <U> Queue<Tuple2<T, U>> zip(Iterable<? extends U> that) {
+        return zipWith(that, Tuple::of);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <U, R> Queue<R> zipWith(Iterable<? extends U> that, BiFunction<? super T, ? super U, ? extends R> mapper) {
         Objects.requireNonNull(that, "that is null");
-        return ofAll(toList().zip((Iterable<U>) that));
+        Objects.requireNonNull(mapper, "mapper is null");
+        return ofAll(toList().zipWith((Iterable<U>) that, mapper));
```

Comment:
It was just a question. I think you are right and it does not compile then. Just leave it as is.